SourceX/controls/plrctrls.cpp

```diff
@@ -166,7 +167,24 @@ bool HasRangedSpell()
 	    && !spelldata[spl].sTownSpell;
 }
 
-void CheckMonstersNearby()
+bool CanTargetMonster(int mi)
+{
+	const auto &monst = monster[mi];
+
+	if (monst._mFlags & (MFLAG_HIDDEN | MFLAG_GOLEM))
+		return false;
+	if (monst._mhitpoints >> 6 <= 0) // dead
+		return false;
+
+	const int mx = monst._mx;
+	const int my = monst._my;
+	if (!(dFlags[mx][my] & BFLAG_LIT)) // not visable
+		return false;
+

```

Comment:
This also used to check `monst.MData->mSelFlag & 7` but no longer does, is that intentional?

Reply:
yes, Golems are the only monster that would fail this test, and they are already weeded out by `MFLAG_HIDDEN | MFLAG_GOLEM`